optimize/functions/functions.go

```diff
@@ -482,6 +482,46 @@ func (Box3D) Minima() []Minimum {
 	}
 }
 
+// BraninHoo implements the Branin-Hoo function. BraninHoo is a 2-dimensional
+// test function for global optimization routines. It has three global minima
```

Comment:
Why for global optimization? Whatever even a local method finds will be a global minimum. Maybe "for gradient-free optimization methods" although the function is smooth and `Grad` could be easily added.